src/protocolsupport/protocol/typeremapper/itemstack/PreFlatteningItemIdData.java

```diff
@@ -46,7 +45,7 @@ public static int getDataFromLegacyCombinedId(int legacyId) {
 	}
 
 	public static int getModernIdByLegacyIdData(int legacyId, int data) {
-		return fromLegacyId.get(formLegacyCombinedId(legacyId, data));
+		return fromLegacyId.getOrDefault(formLegacyCombinedId(legacyId, data), fromLegacyId.getOrDefault(formLegacyCombinedId(legacyId, 0), combinedLegacyStoneId));
```

Comment:
Remove this hack.

Reply:
How then should we get the modern id of damaged items etc?

Reply:
Ok.. but how can we know which material the item is then? Or shall we just remove the legacydata and make complex remappers for wool etc?

Reply:
Hmm, that's true, the complex remapper won't be able to deal with that.
I guess we will just leave it like you suggested for now, and will think about better system later.